homeassistant/components/deconz/cover.py

```diff
@@ -51,92 +49,51 @@ def async_add_cover(lights):
     async_add_cover(gateway.api.lights.values())
 
 
-class DeconzCover(CoverDevice):
+class DeconzCover(DeconzDevice, CoverDevice):
     """Representation of a deCONZ cover."""
 
-    def __init__(self, cover, gateway):
+    def __init__(self, device, gateway):
         """Set up cover and add update callback to get data from websocket."""
-        self._cover = cover
-        self.gateway = gateway
-        self.unsub_dispatcher = None
+        super().__init__(device, gateway)
 
         self._features = SUPPORT_OPEN
         self._features |= SUPPORT_CLOSE
         self._features |= SUPPORT_STOP
         self._features |= SUPPORT_SET_POSITION
 
-    async def async_added_to_hass(self):
-        """Subscribe to covers events."""
-        self._cover.register_async_callback(self.async_update_callback)
-        self.gateway.deconz_ids[self.entity_id] = self._cover.deconz_id
-        self.unsub_dispatcher = async_dispatcher_connect(
-            self.hass, DECONZ_REACHABLE, self.async_update_callback)
-
-    async def async_will_remove_from_hass(self) -> None:
-        """Disconnect cover object when removed."""
-        if self.unsub_dispatcher is not None:
-            self.unsub_dispatcher()
-        self._cover.remove_callback(self.async_update_callback)
-        self._cover = None
-
-    @callback
-    def async_update_callback(self, reason):
-        """Update the cover's state."""
-        self.async_schedule_update_ha_state()
-
     @property
-    def current_cover_position(self):
+    def current_device_position(self):
```

Comment:
This isn't a cover entity state property?

Reply:
Thanks Martin! This is a cover property